packages/lesswrong/client/clientStartup.ts

```diff
@@ -36,3 +42,45 @@ if (document.readyState === 'loading') {
 } else {
   void startupAfterRendering();
 }
+
+// While we are waiting for the main content to be downloaded, we can start importing all deferred components. "Importing"
+// is a bit of a misnomer here, in that `prepareComponent` does some work to memoize and build higher order components.
+// If components are not imported here, they will be imported when they are first used. Importing ahead of time speeds up
+// the frontpage and post page hydration by about 25% (~100ms on an M1 mac).
+//
+// In the event that the main content is downloaded before all components are imported, we want to yield to the main thread
+// to make sure all possible resources are available for rendering the page. The cases where this would occur are:
+// 1. When the main content is served from the cache, in this case the below function doesn't even start importing
+// 2. On a very slow CPU
+const YIELD_MS = 25;
+function importAllComponents() {
+  const componentNames = Object.keys(DeferredComponentsTable);
+  let index = 0;
+
+  function importNextChunk(yieldMs: number | null = null) {
+    const startTime = Date.now();
+
+    while (index < componentNames.length && (!yieldMs || Date.now() - startTime < yieldMs)) {
+      const componentName = componentNames[index];
+
+      // window.ssrRenderedAt being present indicates that the main content has started being parsed.
+      // Yield to the main thread and import all other components as an idle callback (will generally
+      // run >1s later, after the main content has been rendered)
+      if (yieldMs !== null && window.ssrRenderedAt) {
+        requestIdleCallback(() => importNextChunk());
```

Comment:
`requestIdleCallback` doesn't exist on Safari. The page still seems to load (I guess just because the components are imported when they're used), but there's an exception in the console. There's a small polyfill we could use [here](https://github.com/behnammodi/polyfill/blob/master/window.polyfill.js#L7-L24).